src/savegame_json.cpp

```diff
@@ -2354,12 +2320,24 @@ void item::deserialize( JsonIn &jsin )
     const JsonObject data = jsin.get_object();
     io::JsonObjectInputArchive archive( data );
     io( archive );
+    // migration code, used to be std::list<item>
+    if( data.has_array( "contents" ) ) {
+        std::list<item> item_list;
+        optional( data, false, "contents", item_list );
+        for( const item &it : item_list ) {
+            contents.insert_legacy( it );
+        }
+    } else {
+        optional( data, false, "contents", contents );
```

Comment:
```suggestion
        data.read( "contents", contents );
```
Or is there an advantage to using `optional` here?